core/chains/evm/client/errors_test.go

```diff
@@ -405,6 +405,7 @@ func Test_Eth_Errors_Fatal(t *testing.T) {
 		{"failed to forward tx to sequencer, please try again. Error message: 'invalid sender'", true, "Mantle"},
 
 		{"client error fatal", true, "tomlConfig"},
+		{"invalid chain id for signer", true, "Treasure"},
```

Comment:
seems like you have four separate error messages categorized for TreasureFatal, maybe add three others here in the test as well ?

Reply:
Can I just remove those as they were copied from Arbitrum and not relevant actually

Reply:
I see, then you should keep it, as this case is unique for Treasure

Reply:
I've left the `invalid chain id for signer`, and removed the others